File: csharp/0058-length-of-last-word.cs
public class Solution {
    public int LengthOfLastWord(string s) {
        s = s.TrimEnd();
        string[] arr = s.Split(' ');
        return arr[arr.Length - 1].Length;
    }
}

public class NeetCodeSolution {
    public int LengthOfLastWord(string s) {
        var result = 0;
        var i = s.Length - 1;
        while (s[i] == ' ') --i;
        for (; i >= 0; --i) {
            if (s[i] == ' ') {
                return result;
            }
            ++result;
        }
        return result;
    }
}

File: scala/0058-length-of-last-word.scala
object Solution {
    def lengthOfLastWord(s: String): Int = {
    var i = s.length - 1
    var length = 0

    while (s(i) == ' ') {
      i -= 1
    }

    while (i >= 0 && s(i) != ' ') {
      length += 1
      i -= 1
    }

    length
   
    }
}
